modules/standard/List.chpl

```diff
@@ -703,7 +706,9 @@ module List {
       // gasnet/multilocale configurations.
       //
       _appendByRef(x);
+      var insertedElementIndex = _size - 1;
```

Comment:
Let's go ahead and rename `insertedElementIndex` to `result` to stay consistent with the rest of the module.